src/main/java/com/notably/model/block/BlockImpl.java

```diff
@@ -0,0 +1,123 @@
+package com.notably.model.block;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Represents a Block in the Notably data structure.
+ */
+public class BlockImpl implements Block {
```

Comment:
Should we provide a constructor without the `parent` argument? It could be useful for creating the root block.

Reply:
Hmmm... I guess it would make easier to understand the usage too. Will do!